### PR TITLE
Add match logic to AST and Nock.

### PR DESCRIPTION
**Pull request by sigilante**

### CODE REVIEW

File: hoon/lib/jock.hoon [hunk 1]
```diff
@@ -704,6 +713,98 @@
   ?>  (got-punctuator -.tokens end)
   [output +.tokens]
 ::
+++  match-match
+  |*  =tokens
```
Comment (tacryt-socryp) [inline hoon/lib/jock.hoon:717]: I don't think it needs to be wet.

Comment (sigilante) [reply hoon/lib/jock.hoon]: Reverted to dry..

**Comment by sigilante** [comment]
Right now it's fairly aggressive in providing syntax errors, but this clutters the Hoon.

**Comment by sigilante** [comment]
@tacryt-socryp please look at `+hunt`, ll. 1325–1359 in `/hoon/lib/jock.hoon`.

There's a `mint-vain` because I'm not sure what it thinks the head of `jype` can be yet, but I think the recursive Nock for the cell and atom case should work to detect now.

**Comment by sigilante** [comment]
Current status:

- [x] block comments necessary for testing (yakshaving)
- [x] `else` keyword fixed (yakshaving)
- [x] split `match` into `match type` and `match case` switches
- [ ] _in media res_ on match AST return logic
- [ ] _in media res_ on `jype` match logic (`fish-loop` bug, may take a while)

**Comment by sigilante** [comment]
The builder and compiler steps work, but have not been thoroughly tested for correctness yet.  I want to get some test cases in here before merging back into `master`.

The build check fails because it can't find `out.jam`; I suppose I should supply that to the repo.

**Comment by sigilante** [comment]
OK, next need to very carefully check AST and generated Nock.  Once I'm satisfied with that, we can cut a 0.1 release of Jock.

**Comment by sigilante** [comment]
#11 enables testing on this branch.

File: hoon/lib/test-jock.hoon [hunk 2]
```diff
@@ -1,47 +1,90 @@
...
-::/*  dec-jock  %jock  /lib/tests/dec/jock
-::/*  eval  %jock  /lib/tests/eval/jock
-::/*  multi-limb  %jock  /lib/tests/multi-limb/jock
-::/*  compose  %jock  /lib/tests/compose/jock
-::/*  compose-cores  %jock  /lib/tests/compose-cores/jock
-::/*  baby  %jock  /lib/tests/baby/jock
-::/*  comparator  %jock  /lib/tests/comparator/jock
+/*  let-edit              %jock  /lib/tests/let-edit
+/*  let-inner-exp         %jock  /lib/tests/let-inner-exp
+<<<<<<< HEAD
```
Comment (tacryt-socryp) [inline hoon/lib/test-jock.hoon:4]: Git stuff in here

**Review by tacryt-socryp — APPROVED** [review]

**Pull request closed without merging**